src/etc/inc/interfaces.inc

```diff
@@ -1356,6 +1356,80 @@ function interface_gif_configure(&$gif, $gifkey = "") {
 	return $gifif;
 }
 
+/*
+ * $ipsecifnum = get_ipsecifnum($ikeid, $idx, $create=true);
+ * locates or creates and returns an ipsecifnum in the config.
+ */
+function get_ipsecifnum($ikeid, $idx, $create = true) {
+	global $config;
+
+	if (is_array($config['ipsec']['vtimap'])) {
+		$matches = array_filter($config['ipsec']['vtimap'], function($val) use($ikeid, $idx){
+			return ($val['reqid'] == $ikeid and $val['index'] == $idx);
```

Comment:
Each test should be in its own set of parenthesis, plus `and` should probably be `&&`. I don't see anything here that would require using the lower precedence of `and` vs `&&`.

Reply:
In progress
too many config writes, need to move it to vpn_ipsec_phase1.php and vpn_ipsec_phase2.php